components/events-notify.js

```diff
@@ -0,0 +1,73 @@
+import EventsNotifyView from "./events-notify-view";
+import { useEffect, useState } from "react";
+import {PIPE_API} from "../config"
+
+function escapeRegExp(string) {
+  return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"); // $& means the whole matched string
+}
+
+function replaceAll(str, find, replace) {
+  return str.replace(new RegExp(escapeRegExp(find), "g"), replace);
+}
+
+async function notifyEvent(routine, event, args) {
+//   let SERVER_URL = "http://localhost:4000";
+  let request = `${PIPE_API}/execute?event_name=${event}`;
+
+  if (routine !== null && routine !== undefined && routine.length === 2) {
+    request += `&specific_flow_routines={"${routine[0]}": ["${routine[1]}"]}`;
+  }
```

Comment:
Add a comment explaining this maybe?

Reply:
+1

Reply:
No comments. Extra line of code will do the trick to simplify that line.